src/tabbar.cpp

```diff
@@ -127,6 +129,24 @@ void TabBar::setNewTabButton()
     setTabButton(idx, QTabBar::RightSide, Q_NULLPTR);
 }
 
+void TabBar::moveToNextTab()
+{
+    auto index = currentIndex() + 1;
+    if (index == count() - 1) {
+        index = 0; // if we are on last tab, move to first tab (library).
+    }
+    setCurrentIndex(index);
+}
+
+void TabBar::moveToPreviousTab()
+{
+    auto index = currentIndex() - 1;
+    if (index < 0) {
+        index = count() - 1; // if we are on first tab, move to last tab.
+    }
```

Comment:
This relies on the fact that `currentIndex()` returns a signed type. Better rewrite it in a way that would work for an unsigned type too (check before decrementing).

Reply:
In the light of https://github.com/kiwix/kiwix-desktop/pull/867#discussion_r939687653, shouldn't you move from the first tab to the last-but-one tab?

Reply:
> In the light of https://github.com/kiwix/kiwix-desktop/pull/867#discussion_r939687653, shouldn't you move from the first tab to the last-but-one tab?

Correct, that was wrong code but the behaviour was saved by the check in `void TabBar::onCurrentChanged(int index)`

Fixed anyway. Didn't like to be dependent on that check